macos_resources/jenkins-agent.plist

```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>Label</key>
+	<string>org.ros2.ci.REPLACE_HOSTNAME-agent</string>
+	<key>ProgramArguments</key>
+	<array>
+		<string>/usr/bin/java</string>
+		<string>-Djava.awt.headless=true</string>
+		<string>-jar</string>
+		<string>/Users/osrf/jenkins-agent/slave.jar</string>
```

Comment:
I suppose this can't expand env vars by itself ? If it cant, maybe we could also make this a field to replace (REPLACE_HOME) so that it's easily reusable by us (if we rename our users `jenkins` for example) and others. Then I understand you made modular only the machine specific parts and were not aiming to make everything modular so I'm totally fine not acting on this or the CI URL in this PR.

Reply:
Yeah, we can't use environment variables in the argument list. I kept things that were currently universal hard-coded, but I have no problem adding more steps to the setup process to make it more universally applicable.

Happy to make the changes if they're desired.

Reply:
I'd say that fine for now to keep the replace steps to what is strictly necessary. If anybody asks, it's easy to add later on.